enhancements/baremetal/baremetal.md

```diff
@@ -0,0 +1,283 @@
+mentioned earlier, these enhancements rely on the Baremetal Operator (BMO)
+[1] running within the "metal3" pod to manage baremetal hosts. The BMO in
+turn relies on the Ironic service [3] to manage and provision baremetal
+servers.
+
+1. Will enable the user to deploy a control plane with 3 master nodes.
+2. Will enable the user to grow the cluster by dynamically adding worker
+nodes.
+3. Will enable the user to scale down the cluster by removing worker nodes.
+
+### Implementation Details/Notes/Constraints
+
+Baremetal IPI is integrated with OpenShift through the metal3.io [8] project.
+Metal3.io is a set of Kubernetes controllers that wrap the OpenStack Ironic
+project to provide Kubernetes native APIs for managing deployment and
+monitoring of physical hosts.
+
+The installer support for Baremetal IPI deployments is described in more detail
+in [7]. The installer runs on a special "provisioning host" that needs to be
+connected to both a "provisioning network" and an "external network". The
+provisioning network is a dedicated network used just for the purposes of
+configuring baremetal servers to be part of the cluster. The traffic on the
+provisioning network needs to be isolated from the traffic on the external
+network (hence 2 seperate networks.). The external network is used to carry
+cluster traffic which which includes cluster control plane traffic, application
+and data traffic.
+
+Control Plane Deployment
+
+1. A minimin Baremetal IPI deployment consists of 4 hosts, one to be used
+first as a provisioning host and later potentially re-purposed as a worker.
+The other 3 make up the control plane. These 4 hosts need to be connected
+to both the provisioning and external networks.
+
+2. Installation can be kicked off by downloading and running
+"openshift-baremetal-install". This image differs from the "openshift-install"
+binary only because libvirt is needs to be always linked for the baremetal
+install. Removing a bootstrap node would remove the dependency on libvirt
+and then baremetal IPI installs can be part of the normal Openshift installer.
+This is in the roadmap for this work and being investigated.
+
+3. The installer starts a bootstrap VM on the provisioning host. With other
+platform types supported by OpenShift, a cloud already exists and the installer
+runs the bootstrap VM on the control plane of this existing cloud. In the case
+of the baremetal platform type, this cloud does not already exist, so the
+installer starts the bootstrap VM using libvirt.
+
+4. The bootstrap VM needs to be connected to the provisioning network and so the
+the network interface on the provisioning host that is connected to the
+provisioning network needs to be provided to the installer.
+
+5. The bootstrap VM must be configured with a special well-known IP within the
+provisioning network that needs to provided as input to the installer.
+
+6. The installer user Ironic in the bootstrap VM to provision each host that
```

Comment:
when you describe components, list ironic.  Is this running a static pod, is it a daemon on the bootstrap VM, something else?  Does this mean the bootstrap VM has to live past installation?